hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/DiskStartupNetworks.java

```diff
@@ -0,0 +1,322 @@
+import com.hedera.node.app.tss.stores.ReadableTssStore;
+import com.hedera.node.app.tss.stores.ReadableTssStoreImpl;
+import com.hedera.node.config.ConfigProvider;
+import com.hedera.node.config.data.NetworkAdminConfig;
+import com.hedera.node.config.data.TssConfig;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import com.hedera.pbj.runtime.io.stream.ReadableStreamingData;
+import com.hedera.pbj.runtime.io.stream.WritableStreamingData;
+import com.swirlds.config.api.Configuration;
+import com.swirlds.platform.state.service.ReadableRosterStore;
+import com.swirlds.platform.state.service.ReadableRosterStoreImpl;
+import com.swirlds.state.State;
+import com.swirlds.state.lifecycle.StartupNetworks;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.LongUnaryOperator;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * A {@link StartupNetworks} implementation that loads {@link Network} information from a
+ * working directory on disk.
+ */
+public class DiskStartupNetworks implements StartupNetworks {
+    private static final Logger log = LogManager.getLogger(DiskStartupNetworks.class);
+
+    private static final Pattern ROUND_DIR_PATTERN = Pattern.compile("\\d+");
+
+    public static final String ARCHIVE = ".archive";
+    public static final String GENESIS_NETWORK_JSON = "genesis-network.json";
+    public static final String OVERRIDE_NETWORK_JSON = "override-network.json";
+
+    private final long selfNodeId;
+    private final ConfigProvider configProvider;
+    private final TssBaseService tssBaseService;
+
+    private boolean isArchived = false;
+
+    public DiskStartupNetworks(
+            final long selfNodeId,
+            @NonNull final ConfigProvider configProvider,
+            @NonNull final TssBaseService tssBaseService) {
+        this.selfNodeId = selfNodeId;
+        this.configProvider = requireNonNull(configProvider);
+        this.tssBaseService = tssBaseService;
+    }
+
+    @Override
+    public Network genesisNetworkOrThrow() {
+        return loadNetwork(configProvider.getConfiguration(), GENESIS_NETWORK_JSON)
+                .orElseThrow(() -> new IllegalStateException("Genesis network not found"));
+    }
+
+    @Override
+    public Optional<Network> overrideNetworkFor(final long roundNumber) {
+        final var config = configProvider.getConfiguration();
+        final var unscopedNetwork = loadNetwork(config, OVERRIDE_NETWORK_JSON);
+        if (unscopedNetwork.isPresent()) {
+            return unscopedNetwork;
+        }
+        return loadNetwork(config, "" + roundNumber, OVERRIDE_NETWORK_JSON);
+    }
+
+    @Override
+    public void setOverrideRound(final long roundNumber) {
```

Comment:
When would we need this? Wouldn't all the cases have a round number they start from?

Reply:
I _think_ we need this to mark the earliest round number that we used an override network for....but it's hard to say without a live test. Will examine further once `PlatformStateService` restart schema is done. 🤔